src/mbgl/text/collision_tile.cpp

```diff
@@ -156,49 +148,53 @@ Box CollisionTile::getTreeBox(const Point<float>& anchor, const CollisionBox& bo
     };
 }
 
-std::vector<IndexedSubfeature> CollisionTile::queryRenderedSymbols(const GeometryCoordinates& queryGeometry, const float scale) {
-
+std::vector<IndexedSubfeature> CollisionTile::queryRenderedSymbols(const GeometryCoordinates& queryGeometry, float scale) {
     std::vector<IndexedSubfeature> result;
-    if (queryGeometry.empty()) return result;
-
-    std::unordered_map<std::string, std::unordered_set<std::size_t>> sourceLayerFeatures;
+    if (queryGeometry.empty() || (tree.empty() && ignoredTree.empty())) {
+        return result;
+    }
 
-    mapbox::geometry::multi_point<float> rotatedPoints {};
-    rotatedPoints.reserve(queryGeometry.size());
-    std::transform(queryGeometry.cbegin(), queryGeometry.cend(), std::back_inserter(rotatedPoints),
+    // Generate a rotated query box out of the original query geometry.
+    mapbox::geometry::multi_point<float> rotatedQuery {};
+    rotatedQuery.reserve(queryGeometry.size());
+    std::transform(queryGeometry.cbegin(), queryGeometry.cend(), std::back_inserter(rotatedQuery),
                    [&](const auto& c) { return util::matrixMultiply(rotationMatrix, convertPoint<float>(c)); });
-    const auto box = mapbox::geometry::envelope(rotatedPoints);
+    const auto box = mapbox::geometry::envelope(rotatedQuery);
 
-    const auto& anchor = box.min;
-    CollisionBox queryBox(anchor, 0, 0, box.max.x - box.min.x, box.max.y - box.min.y, scale);
-    auto predicates = bgi::intersects(getTreeBox(anchor, queryBox));
+    // Account for scaling when generating the query box.
+    const float x2 = (box.max.x - box.min.x) * scale;
+    const float y2 = (box.max.y - box.min.y) * scale;
+    const auto queryBox = CollisionBox { box.min, 0, 0, x2, y2, scale };
```

Comment:
> Why do these units need to be multiplied by scale?

My suspicion is that we must follow the same approach done when populating the symbol buckets:

https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/layout/symbol_layout.cpp#L422-L434

You can see that placementZoom is actually `zoom` - [which is an integer zoom value](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/style/layers/symbol_layer_impl.cpp#L70) multiplied the scale difference.

> Also, why would the width/height be multiplied but not the point box.min?

Anchor (box.min) doesn't get multiplied by scale when generating the tree box:
https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/text/collision_tile.cpp#L149-L154

Because `(x1, y1)` is always 0, we don't need them to scale.